undofx/src/main/java/org/fxmisc/undo/UndoManagerFactory.java

```diff
@@ -14,118 +15,249 @@
 
 public interface UndoManagerFactory {
 
-    <C> UndoManager create(
+    /**
+     * Creates an {@link UndoManager} where changes will never be merged and every change emitted from the
+     * changeStream is considered to be a non-identity change.
+     *
+     * @param changeStream the {@link EventStream} that emits changes
+     * @param invert inverts the change
+     * @param apply applies the change (undo/redo)
+     * @param <C> the type of change object to use
+     */
+    default <C> UndoManager create(
+            EventStream<C> changeStream,
+            Function<? super C, ? extends C> invert,
+            Consumer<C> apply) {
+        return create(changeStream, invert, apply, (c1, c2) -> Optional.empty());
+    }
+
+    /**
+     * Creates an {@link UndoManager} where every change emitted from the changeStream is considered to be a
+     * non-identity change
+     *
+     * @param changeStream the {@link EventStream} that emits changes
+     * @param invert inverts the change
+     * @param apply applies the change (undo/redo)
+     * @param merge merges two changes into one change. If the resulting change is an identity change, the two
+     *              changes will not be merged.
```

Comment:
This comment is already outdated and lying. This is the trouble with stringly typed documentation: gets out of sync with code so easily.